test/pjrt/test_ddp.py

```diff
@@ -0,0 +1,37 @@
+from absl.testing import absltest, parameterized
+import torch.distributed as dist
+import torch.nn as nn
+from torch.nn.parallel import DistributedDataParallel as DDP
+import torch_xla.core.xla_model as xm
+import torch_xla.distributed.xla_backend
+from torch_xla.experimental import pjrt
+
+
+def _init_xla_backend(init_file: str):
+  rank = xm.get_ordinal()
+  world_size = xm.xrt_world_size()
+
+  dist.init_process_group(
+    "xla",
+    init_method=f"file://{init_file}",
+    rank=rank,
+    world_size=world_size)
+
+class TestPjRtDistributedDataParallel(parameterized.TestCase):
+
+  @staticmethod
+  def _ddp_init(init_file: str):
+    _init_xla_backend(init_file)
+
+    device = xm.xla_device()
```

Comment:
Do we need to specify the device ordinal here? I guess PJRT takes care of mapping the default xla device to the actual physical TPU core. And then during the lowering, the XLA IR will be able to extract the correct physical device ordinal.

Reply:
Correct. If I omit the `to(device)` and pass `device_ids` into DDP I get this error:

```
ValueError: DistributedDataParallel device_ids and output_device arguments only work with single-device/multiple-device GPU modules or CPU modules, but got device_ids [1], output_device None, and module parameters {device(type='cpu')}.
```

Reply:
Oh, `device_ids` is more or less a CUDA concept: https://github.com/pytorch/pytorch/blob/master/torch/nn/parallel/distributed.py#L436.

What I was referring to is something possibly like `device = xm.xla_device(rank)`.

Reply:
Gotcha. The initialization in `xmp.spawn` for XRT and `pjrt.run_multiprocess` in PJRT take care of assigning the default device to each process/thread such that `xm.xla_device()` will always return the device assigned to a replica.

Reply:
Sounds good. Thank you, Will.